src/lib/main.js

```diff
@@ -128,68 +128,19 @@ module.exports = async (argv, env) => {
   //
   // Build up actions
   //
-  let actions = cmdArgs._;
-  if (actions.length) {
-    const unsupported = actions.filter(a => !supportedActions.includes(a));
-    if(unsupported.length) {
-      throw new Error(`Unsupported action(s): ${unsupported.join(' ')}`);
-    }
-  } else {
-    actions = !cmdArgs.archive ? defaultActions : defaultArchiveActions;
-  }
-
-  if (cmdArgs['skip-git-check']) {
-    actions = actions.filter(a => a !== 'check-git');
-  }
-
   const skipValidate = cmdArgs['skip-validate'];
-  if(skipValidate) {
-    warn('Skipping all form validation.');
-    const validateActions = [
-      'validate-app-forms',
-      'validate-collect-forms',
-      'validate-contact-forms'
-    ];
-    actions = actions.filter(action => !validateActions.includes(action));
-  } else {
-    const addFormValidationIfNecessary = (formType) => {
-      const updateFormsIndex = actions.indexOf(`upload-${formType}-forms`);
-      if (updateFormsIndex >= 0 && actions.indexOf(`validate-${formType}-forms`) < 0) {
-        actions.splice(updateFormsIndex, 0, `validate-${formType}-forms`);
-      }
-    };
-    addFormValidationIfNecessary('app');
-    addFormValidationIfNecessary('collect');
-    addFormValidationIfNecessary('contact');
-  }
-
-  actions = actions.map(actionName => {
-    const action = require(`../fn/${actionName}`);
-
-    if (typeof action.execute !== 'function') {
-      throw new Error(`${actionName} has not been implemented correctly: no 'execute' function`);
-    }
-
-    if (!Object.hasOwnProperty.call(action, 'requiresInstance')) {
-      action.requiresInstance = true;
-    }
-
-    action.name = actionName;
-
-    return action;
-  });
+  const actions = buildActions(cmdArgs, skipValidate);
 
   //
   // Initialize the environment
   //
   const projectName = fs.path.basename(pathToProject);
 
-  const apiUrl = getApiUrl(cmdArgs, env);
   const requiresInstance = actions.some(action => action.requiresInstance);
-  if (requiresInstance) {
-    if (!apiUrl) {
-      throw new Error('Failed to obtain a url to the API');
-    }
+  const apiUrl = requiresInstance && getApiUrl(cmdArgs, env);
+  if (requiresInstance && !apiUrl) {
+    usage();
+    throw new Error('Failed to obtain a url to the API');
   }
```

Comment:
Is it even possible to hit this case now? I think `getApiUrl` always return a value or throw an exception right?

Reply:
i don't believe it's possible as-written.